src/tests/wasm.rs

```diff
@@ -88,6 +89,96 @@ fn load_from_file(filename: &str) -> Module {
     Module::from_buffer(wasm_buf).unwrap()
 }
 
+#[test]
+fn loader_on_inc_i32() {
+    // Name of function contained in WASM file (note the leading underline)
+    const FUNCTION_NAME: &str = "_inc_i32";
+    // The WASM file containing the module and function
+    const WASM_FILE: &str = &"res/fixtures/inc_i32.wast";
+
+    let mut module = load_from_file(WASM_FILE);
+
+    // To simulate absent function bodies we extract the code and clearing code map,
+    // so that functions will be created without bodies during module instantiation.
+    let code_map = module.code_map.clone();
+    module.code_map.clear();
+
+    let env = Env::new();
+
+    let instance = ModuleInstance::new(&module, &ImportsBuilder::new().with_resolver("env", &env))
+        .expect("Failed to instantiate module")
+        .assert_no_start();
+
+    let i32_val = 42;
+    // the functions expects a single i32 parameter
+    let args = &[RuntimeValue::I32(i32_val)];
+    let exp_retval = Some(RuntimeValue::I32(i32_val + 1));
+
+    struct Loader<'a> {
+        bodies: &'a [parity_wasm::elements::FuncBody],
+        code_map: Vec<Instructions>,
+    }
+
+    impl<'a> runner::Loader for Loader<'a> {
+        fn load_function_body(&self, index: usize) -> Option<crate::func::FuncBody> {
+            println!("Loading function body index {}", index);
+
+            // In a real setup these should be loaded from an external resource
+            let locals = self.bodies.get(index)?.locals().to_vec();
+            let code = self.code_map.get(index)?;
+
+            // Allocating body filled with placeholder instructions to be patched later
+            let len = code.instructions().len();
+            let mut stub = Instructions::with_capacity(len);
+            for _ in 0..len {
+                stub.push(crate::InstructionInternal::NotLoaded);
+            }
```

Comment:
Shouldn't the whole code be replaced with a single `NotLoaded` instruction? Otherwise what is the point of stubbing it when it is replaced with a code of the same size.